src/Application/Common/Contracts/IRequestHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Threading.Tasks;
+
+namespace Codidact.Core.Application.Common.Contracts
+{
+    /// <summary>
+    /// Defines a handler for a request
+    /// </summary>
+    /// <typeparam name="TRequest">The type of request being handled</typeparam>
+    /// <typeparam name="TResponse">The type of response from the handler</typeparam>
+    public interface IRequestHandler<in TRequest, TResponse>
```

Comment:
A request handler should be allowed to return a type that derives from `TResponse`. (I have not tested this.)

```suggestion
    public interface IRequestHandler<in TRequest, out TResponse>
```

Reply:
Actually this might be correct as it is, it depends how the serialization stuff is implemented.

Reply:
Its about the async returning only Task.
It was originally like you wrote but I changed it because I want to return Task<TResult> and not TResult